csc-services/controllers/impl/DashboardControllerImpl.js

```diff
@@ -76,7 +83,36 @@ class DashboardControllerImpl extends WebcController {
 		});
 	}
 
-	async getNotificationDataForStatus(data) {
+
+	handleShipmentMessages(){
+		this.communicationService.listenForMessages(async (err, data) => {
+			if (err) {
```

Comment:
The notification will reach this point only if the ShipmentService will be ready before OrderService.
The idea is that you can have only one listener `this.communicationService.listenForMessages` because it will answer to the first one who is registered.
The approach seems to be ok, but the solution is not.
You need to make the separation of the messages within the same listener.